tests/testsuite/out_dir.rs

```diff
@@ -281,6 +281,29 @@ fn cargo_build_out_dir() {
     );
 }
 
+#[cargo_test]
+fn unsupported_short_out_dir_flag() {
+    let p = project()
+        .file("src/main.rs", r#"fn main() { println!("Hello, World!") }"#)
+        .build();
+
+    p.cargo("build -Z unstable-options -O")
+        .masquerade_as_nightly_cargo(&["out-dir"])
+        .with_stderr(
+            "\
+error: unexpected argument '--unsupported-short-out-dir-flag' found
```

Comment:
This seems pretty odd as user never type this. Do we have a way to avoid showing`--unsupported-short-out-dir-flag`?

Reply:
I assume this is because of `flag` which can instead be bypassed.

Reply:
I assume this is because of `flag` which can instead be bypassed.

Reply:
Yes, from my Additional information in PR body.
> I am not sure if error: unexpected argument '--unsupported-short-out-dir-flag' found makes sense.
> 
> I don't know how to make it show error: unexpected argument '-O'

Reply:
Thanks! Updated.